lib/components/position_component.dart

```diff
@@ -45,8 +84,22 @@ abstract class PositionComponent extends Component {
     height = size.y;
   }
 
-  Rect toRect() => Rect.fromLTWH(x - anchor.relativePosition.dx * width,
-      y - anchor.relativePosition.dy * height, width, height);
+  /// Returns the size of this component starting at (0, 0).
+  /// Effectively this is it's position with respect to itself.
+  /// Use this if the canvas is already translated by (x, y).
+  Rect toZeroRect() => Rect.fromLTWH(0, 0, width, height);
```

Comment:
This name is confusing imho, sounds like a zero width+length rectangle. Maybe toOriginRect? Or maybe origin is "defined" by the anchor?